TouchPortalAPI/sdk_tools.py

```diff
@@ -432,14 +434,16 @@ def validateDefinitionObject(data:dict):
     _validateDefinitionDict(data, TPSDK_ATTRIBS_ROOT, sdk_v)
     return len(g_messages) == 0
 
-def validateDefinitionString(data:str):
+def validateDefinitionString(data):
     """
     Validates a TP plugin definition structure from JSON string.
     `data` is an entry.tp JSON string
     Returns `True` if no problems were found, `False` otherwise.
     Use `getMessages()` to check for any validation warnings which may be generated.
     """
-    return validateDefinitionObject(json.loads(data))
+    if isinstance(data, str):
```

Comment:
Is this change actually being used somewhere?

Reply:
Yes. here is where its used. https://github.com/KillerBOSS2019/TouchPortal-API/blob/3dafd6df625b614aea40d8d83936ad4373f6f99c/TouchPortalAPI/tppdoc.py#L351

I had to do this because `generateDefinitionFromScript` generates string. because `validateDefinitionObject` when start it tries to parse it already. so I have to use json.loads before calling that method

Reply:
> because validateDefinitionObject when start it tries to parse it already

I'm confused... what do you mean by that?  `validateDefinitionObject(generateDefinitionFromScript(...))` doesn't work?

This method is called `validateDefinitionString()` so it's just strange that it would take something that's _not_ a string.  If `validateDefinitionObject()` doesn't work for your needs for some reason, then fix that or create a new method.